### PR TITLE
John conroy/organ samples polish

### LINKED ISSUE #2343: Samples count subhead

### LINKED ISSUE #2344: Add to list button

### PR DESCRIPTION
**Pull request by john-conroy**
Towards #2182. Fix #2344. Fix #2343. Tests, stories and a little further cleanup in #2330.

Organ page samples table

- Adds samples count sub-header.
- Updates button text from 'View All Samples' to 'View Data on Search Page'.
- Enables users to save samples directly from table.

Saved entities
- Modifies saved entities to only store an entity's UUID and saved timestamp. Additional data for the saved entities table on the saved lists and saved list pages is now fetched asynchronously and protected will not be displayed while logged out.
- Text in buttons and modals switched from 'Add to' to 'Add to List'.

New reusable components
- 'Selectable' table components.
-  Loading table rows.

Screen Shot 2022-01-07 at 3 55 08 PM

Screen Shot 2022-01-07 at 3 55 17 PM

Screen Shot 2022-01-07 at 3 55 34 PM

Screen Shot 2022-01-13 at 5 06 57 PM

Screen Shot 2022-01-13 at 5 07 25 PM

Before

Screen Shot 2022-01-10 at 3 45 48 PM

### CODE REVIEW

**Comment by mccalluc** [comment]
@john-conroy -- If they haven't already been elevated to issues, can you make issues from the points on #2182 that this PR will address, and link them here? I'm trying to get a handle on how much still needs to be done, outside this PR. Thanks!

**Comment by mccalluc** [comment]
Double check with Tiffany about the button style -- At one point we wanted the samples icon on there, but we may have decided to go in a different direction. Now looking at the code...

File: context/app/static/js/shared-styles/tables/LoadingTableRows/LoadingTableRows.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import Skeleton from '@material-ui/lab/Skeleton';
+
+function LoadingTableRows({ numberOfRows, numberOfCols }) {
+  return [...Array(numberOfRows).keys()].map(() => (
+    <TableRow>
+      {[...Array(numberOfCols).keys()].map(() => (
+        <TableCell>
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/tables/LoadingTableRows/LoadingTableRows.jsx:10]:
```suggestion

function range(n) {
  return [...Array(numberOfRows).keys()];
}

function LoadingTableRows({ numberOfRows, numberOfCols }) {
  return range(numberOfRows).map(() => (
    <TableRow>
      {range(numberOfCols).map(() => (
        <TableCell>
```

**Review by mccalluc — COMMENTED** [review]
Looks fine. If it's not hard, could you add a couple tests around the utilties... Maybe `LoadingTableRows` and something else where there's a little bit of logic? Few scattered comments that you can take or leave.

**Comment by mccalluc** [comment]
... and it looks like you've covered my comments in the follow-up PR, [for example](https://github.com/hubmapconsortium/portal-ui/pull/2330/files#diff-480767496ea624dcce9b7a7843adbd721438d9b7ce16f5ebf34d6dbb4cf34aa1R13). Sorry to bother.

File: context/app/static/js/components/organ/Samples/Samples.jsx [hunk 2]
```diff
@@ -25,10 +35,9 @@ const columns = [
   { id: 'last_modified_timestamp', label: 'Last Modified' },
 ];
 
-function Samples(props) {
-  const { searchTerms } = props;
+function Samples({ searchTerms }) {
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/organ/Samples/Samples.jsx:38]: It's more concise and is a fixed location, so I think so.

File: context/app/static/js/components/savedLists/SavedEntitiesTable/SavedEntitiesTable.jsx [hunk 3]
```diff
@@ -101,35 +89,42 @@ function SavedEntitiesTable({ savedEntities, deleteCallback, setShouldDisplaySav
...
+              {isLoading ? (
+                <LoadingTableRows numberOfRows={Object.keys(savedEntities).length} numberOfCols={5} />
+              ) : (
+                searchHits.map(
+                  ({ _id, _source: { hubmap_id, group_name, entity_type } }, i) =>
+                    _id in savedEntities && ( // on item deletion savedEntites will update before searchHits
+                      <TableRow key={_id}>
+                        <SelectableRowCell rowKey={_id} index={i} />
+                        <TableCell>
+                          <LightBlueLink href={`/browse/${hubmap_id}`}>{hubmap_id}</LightBlueLink>
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/savedLists/SavedEntitiesTable/SavedEntitiesTable.jsx]: Done, thanks!

File: context/app/static/js/components/organ/Samples/Samples.jsx [hunk 4]
```diff
@@ -59,44 +68,67 @@ function Samples(props) {
...
+                })
+                .map(({ _id: uuid, _source: { hubmap_id, donor, descendant_counts, last_modified_timestamp } }, i) => (
+                  <TableRow key={uuid}>
+                    <SelectableRowCell rowKey={uuid} index={i} />
+                    <TableCell>
+                      <LightBlueLink href={`/browse/sample/${uuid}`} variant="body2">
+                        {hubmap_id}
+                      </LightBlueLink>
+                    </TableCell>
+                    <TableCell>{donor?.mapped_metadata?.age_value}</TableCell>
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/organ/Samples/Samples.jsx]: I've added `age_unit` and a util to create the string including both.

File: context/app/static/js/components/organ/Samples/Samples.jsx [hunk 5]
```diff
@@ -59,44 +68,67 @@ function Samples(props) {
...
+          </div>
+        }
         buttons={
-          <Button color="primary" variant="contained" component="a" href={searchUrl}>
-            <StyledSampleIcon />
-            View All Samples
-          </Button>
+          <>
+            <Button color="primary" variant="outlined" component="a" href={searchUrl}>
+              View Data on Search Page
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/organ/Samples/Samples.jsx:81]: We decided to update all of these buttons seen on detail pages for derived entities to the outlined version seen here.

**Pull request merged**